src/commitlint/linter/_linter.py

```diff
@@ -47,16 +54,30 @@ def lint_commit_message(
         console.verbose("commit message ignored, skipping lint")
         return True, []
 
+    validator_instances: List[CommitValidator] = []
+
+    if not disable_max_header_length:
+        validator_instances.append(
+            HeaderLengthValidator(
+                commit_message=commit_message,
+                **{"max_header_length": max_header_length},  # type: ignore
+            )
+        )
+
     # for skip_detail check
     if skip_detail:
         console.verbose("running simple validators for linting")
+
+        validator_instances.append(
+            SimplePatternValidator(commit_message=commit_message)
+        )
+
         return run_validators(
-            commit_message,
-            validator_classes=[HeaderLengthValidator, SimplePatternValidator],
+            validator_classes=validator_instances,
             fail_fast=True,
         )
 
+    validator_instances.append(PatternValidator(commit_message=commit_message))
+
     console.verbose("running detailed validators for linting")
-    return run_validators(
```

Comment:
suggestion: I never thought the task was going to be this tough. I have a few alternative suggestions that are effective and require fewer code changes.

However, our tool has a config for storing such configurations. It's already been implemented for verbose and quiet options, you can check that. With that said, the changes will only be on `HeaderLengthValidator` where the actual validator runs.